engine/src/memory_engine.rs

```diff
@@ -206,4 +308,18 @@ mod test {
             .collect::<Vec<(Vec<u8>, Vec<u8>)>>();
         assert_eq!(res_3.sort(), expected_all_values.sort());
     }
+
+    #[test]
+    fn snapshot_should_work() {
+        let engine = MemoryEngine::new(&TESTTABLES).unwrap();
+        let put = WriteOperation::new_put("kv", "key", "value");
+        assert!(engine.write_batch(vec![put], false).is_ok());
+
+        let snapshot = engine.get_snapshot("", &TESTTABLES).unwrap();
+        let engine_2 = MemoryEngine::new(&TESTTABLES).unwrap();
```

Comment:
On test coverage grounds, I think we can put another key-value pair into the original engine, and make sure that the new put operation after taking a snapshot won't show up in engine_2.